types/networks/src/lib.rs

```diff
@@ -242,19 +180,18 @@ impl ChainConfig {
     }
 
     pub fn network_version(&self, epoch: ChainEpoch) -> NetworkVersion {
-        let height = self
-            .height_infos
+        let height_infos = &mut self.height_infos.clone();
+        normalize(height_infos);
+        let height = height_infos
             .iter()
             .rev()
             .find(|info| epoch > info.epoch)
             .map(|info| info.height)
             .unwrap_or(Height::Breeze);
 
-        self.version_schedule
-            .iter()
-            .find(|info| height == info.height)
-            .map(|info| info.version)
-            .expect("A network version should exist even if not specified in the config (a default exists).")
+        // unfallible as it is guaranteed that number of heights
```

Comment:
where is the static check that guarantees this?

Reply:
Added a compile time check for this